### PR TITLE
feature/RAD-172-update-dropwizard-external-auth-bundle-library-dependencies

### PR DESCRIPTION
**Pull request by adam-kimpton**
- Updated dependencies
- Upgraded to Java 17
- Upgraded to Gradle 8
- Move to use Maven Central rather than JCenter

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,14 +117,14 @@ To create your own instances of `ExternallyAuthenticatedAuthFilterFactory` from
 
 2. Add the full class name of your subclass of `ExternallyAuthenticatedAuthFilterFactory` inside.
 
-# Development
+## Development
 
 - Publish new versions via:
 
         export BINTRAY_USER=<user>
```
Comment (adam-kimpton) [inline README.md:124]: @al-jeyapal this needs to be changed to however publishing to maven central works as bintray (along with jcenter) has become depricated and read-only.

Comment (al-jeyapal) [reply README.md:124]: @adam-kimpton yep, look at the following:
https://github.com/brightsparklabs/asanti/blob/ea69e8ff81eb46c1f26643219e77b74398ee3d5f/.github/workflows/publish.yml#L36

Basically set those 4 env vars, then call `./gradlew publishToMavenCentral`

File: build.gradle [hunk 2]
```diff
@@ -7,52 +7,76 @@
...
+
+/*
+ * Maven Central requires artifacts to be released via Sonatype Nexus.
+ * The Nexus Publish Plugin can only be applied to the root project.
+ * Do not apply it if this project is being included as a subproject.
+ */
+def isRootProject = project.path == ':'
+if (isRootProject) {
+    apply plugin: 'io.github.gradle-nexus.publish-plugin'
 }
```
Comment (al-jeyapal) [inline build.gradle:47]: You can get rid of this check. It only applies to gradle projects which have subprojects.
So just include the plugin in the `plugins` block and get rid of the buildscript.

File: build.gradle [hunk 3]
```diff
@@ -7,52 +7,76 @@
...
 // PROJECT IDENTIFICATION
 // -----------------------------------------------------------------------------
 
 project.description  = 'Dropwizard bundle for trusting externally authenticated users'
 project.ext.url      = 'https://github.com/brightsparklabs/dropwizard-external-auth-bundle'
-project.ext.scm      = 'https://github.com/brightsparklabs/dropwizard-external-auth-bundle.git'
-
+project.ext.scm      = 'scm:git@github.com:brightsparklabs/dropwizard-external-auth-bundle.git'
+project.group       = 'com.brightsparklabs'
+project.version     = 'git describe --always --dirty'.execute().text.trim()
```
Comment (al-jeyapal) [inline build.gradle:60]: You shouldn't need to set `project.version` or `project.group`, the BSL baseline plugin will do it for you.

File: build.gradle [hunk 4]
```diff
@@ -74,12 +98,12 @@ dependencies {
 
     testImplementation(
             // Use the awesome Spock testing and specification framework even with Java
-            'org.codehaus.groovy:groovy-all:2.5.6',
-            'org.spockframework:spock-core:1.2-groovy-2.5',
-            'junit:junit:4.12',
+            //            'org.codehaus.groovy:groovy-all:3.0.17',
```
Comment (al-jeyapal) [inline build.gradle:101]: Why is this commented out?

File: build.gradle [hunk 5]
```diff
@@ -74,12 +98,12 @@ dependencies {
...
+            'org.spockframework:spock-core:2.4-M1-groovy-4.0',
+            'junit:junit:4.13.2',
 
             // So that we can suppress some spotbugs reports
-            'com.github.spotbugs:spotbugs-annotations:3.1.12',
+            'com.github.spotbugs:spotbugs-annotations:4.7.3',
```
Comment (al-jeyapal) [inline build.gradle:106]: We can get rid of spotbugs, we've moved to errorprone.

File: build.gradle [hunk 6]
```diff
@@ -74,12 +98,12 @@ dependencies {
 
     testImplementation(
             // Use the awesome Spock testing and specification framework even with Java
-            'org.codehaus.groovy:groovy-all:2.5.6',
-            'org.spockframework:spock-core:1.2-groovy-2.5',
-            'junit:junit:4.12',
+            //            'org.codehaus.groovy:groovy-all:3.0.17',
+            'org.spockframework:spock-core:2.4-M1-groovy-4.0',
+            'junit:junit:4.13.2',
```
Comment (al-jeyapal) [inline build.gradle:103]: Can we not go to JUnit 5?

File: build.gradle [hunk 7]
```diff
@@ -94,82 +118,126 @@ dependencies {
 dependencies {
 }
```
Comment (al-jeyapal) [inline build.gradle:119]: Get rid of this, we just defined dependencies above.

File: build.gradle [hunk 8]
```diff
@@ -94,82 +118,126 @@ dependencies {
...
+    // Use mavenCentral for resolving dependencies.
+    mavenCentral()
 }
 
-// Include javadoc and sources (requirements for publishing to Maven Central)
-task javadocJar(type: Jar) {
-    classifier = 'javadoc'
-    from javadoc
-}
+dependencies {
```
Comment (al-jeyapal) [inline build.gradle:131]: Why do we have yet another `dependencies` block?
There should only be one.

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/ExternallyAuthenticatedAuthFilterFactory.java [hunk 9]
```diff
@@ -55,8 +58,8 @@ public abstract class ExternallyAuthenticatedAuthFilterFactory implements Discov
     // -------------------------------------------------------------------------
 
     /**
-     * @return The field in the MDC (Mapped Diagnostic Context) that the authenticated user's
-     *     username is stored against. Default: {@value DEFAULT_MDC_USERNAME_FIELD}.
+     * Returns The field in the MDC (Mapped Diagnostic Context) that the authenticated user's
+     * username is stored against. Default: {@value DEFAULT_MDC_USERNAME_FIELD}.
```
Comment (al-jeyapal) [inline src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/ExternallyAuthenticatedAuthFilterFactory.java:62]: You still need the `@return` comment as well, its part of javadoc.
Our practice is to either duplicate the statement. Or allow empty javadopc descriptions in errorprone config, and just use `@return`.

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AuthenticationDeniedException.java [hunk 10]
```diff
@@ -17,14 +17,30 @@
  */
 public class AuthenticationDeniedException extends Exception {
 
+    /**
+     * Constructs a new authentication denied exception with the specified detail message.
+     *
+     * @param message the detail message.
+     */
```
Comment (al-jeyapal) [inline src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AuthenticationDeniedException.java:24]: Thanks for adding these.

**Comment by al-jeyapal** [comment]
Resolves #8 and #9

**Pull request merged**